### PR TITLE
sql: report an error for unsupported types

### PR DESCRIPTION
**Pull request by sergos**
SQL dumps core in case select delivers some of unsupported types - e.g. UUID or DECIMAL.
I fixed it with reporting an error of unsupported type.

### CODE REVIEW

File: src/box/sql/vdbemem.c [hunk 1]
```diff
@@ -449,7 +449,8 @@ mem_mp_type(struct Mem *mem)
 		     mem->subtype != SQL_SUBTYPE_MSGPACK)
 			return MP_BIN;
 		assert(mp_typeof(*mem->z) == MP_MAP ||
-		       mp_typeof(*mem->z) == MP_ARRAY);
+		       mp_typeof(*mem->z) == MP_ARRAY ||
+                       mp_typeof(*mem->z) == MP_EXT);
```
Comment (tsafin) [inline src/box/sql/vdbemem.c:453]: Spaces used, instead of tabs for indentation

File: src/box/sql/vdbe.c [hunk 2]
```diff
@@ -382,7 +382,8 @@ mem_apply_type(struct Mem *record, enum field_type type)
 		if ((record->flags & MEM_Subtype) != 0 &&
 		    record->subtype == SQL_SUBTYPE_MSGPACK) {
 			assert(mp_typeof(*record->z) == MP_MAP ||
-			       mp_typeof(*record->z) == MP_ARRAY);
+			       mp_typeof(*record->z) == MP_ARRAY ||
+                               mp_typeof(*record->z) == MP_EXT);
```
Comment (tsafin) [inline src/box/sql/vdbe.c:386]: Spaces, instead of tabs

File: test/sql/gh-5913-error-on-unsupported-type.result [hunk 3]
```diff
@@ -0,0 +1,87 @@
+-- test-run result file version 2
```
Comment (tsafin) [inline test/sql/gh-5913-error-on-unsupported-type.result:1]: This is very personal, and not obligatory, but I believe it's better to create new tests inside of test/sql-tap than test/sql with result files. Could you please convert them to sql-tap for easier debug?

**Review by tsafin — APPROVED** [review]

**Comment by Korablev77** [comment]
Still crashes:
```
tarantool> box.execute('update T set "dec" = 3')
tarantool: /home/nikita/tarantoolOrigin/tarantool/src/box/sql/vdbe.c:433: mem_is_type_compatible: Assertion `mp_type < MP_EXT' failed.
Aborted (core dumped)
```

File: src/box/sql/vdbemem.c (hunk 1 above)
Comment (sergos) [reply src/box/sql/vdbemem.c]: fixed

File: src/box/sql/vdbe.c (hunk 2 above)
Comment (sergos) [reply src/box/sql/vdbe.c]: fixed

File: test/sql/gh-5913-error-on-unsupported-type.result (hunk 3 above)
Comment (sergos) [reply test/sql/gh-5913-error-on-unsupported-type.result]: I totally agree, rewrote to TAP

**Review by tsafin — APPROVED** [review]

**Comment by Korablev77** [comment]
At least you should handle these types in built-in functions:
```
tarantool> box.execute('select round("dec") from t')
---
- metadata:
  - name: COLUMN_1
    type: integer
  rows:
  - [0]
...
tarantool> box.execute('select max("uid") from t')
---
- null
- SQL does not support scalar
```
I haven't checked all functions, but some of them may also crash.

**Comment by pgulutzan** [comment]
For issue#6164 there was a problem using QUOTE(uuid-value), which has been fixed (though the fix is not pushed yet).
But this still causes a segmentation fault:
box.schema.space.create('tester')
box.space.tester:format({{'a',type='number'},{'b',type='any',is_nullable=true}})
box.space.tester:create_index('a')
box.space.tester:insert{0,{5}}
box.execute([[SELECT QUOTE("b") FROM "tester";]])
So this is a request: please test with QUOTE, which for some reason seems more dangerous than other functions.

**Pull request closed without merging**